apis/projects/v1alpha1/deploykey_type.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeployKeyParameters define desired state of Gitlab Deploy Key
+// https://docs.gitlab.com/ee/api/deploy_keys.html
+// At least 1 of [ProjectId, ProjectIdRef, ProjectIdSelector] required
+type DeployKeyParameters struct {
+	// The ID or URL-encoded path of the project owned by the authenticated user
+	// +optional
+	// +immutable
+	ProjectId *string `json:"projectId,omitempty"`
+
+	// ProjectIDRef is a reference to a project to retrieve its projectId
+	// +optional
+	// +immutable
+	ProjectIdRef *xpv1.Reference `json:"projectIdRef,omitempty"`
+
+	// ProjectIDSelector selects reference to a project to retrieve its projectId.
+	// +optional
+	// +immutable
+	ProjectIdSelector *xpv1.Selector `json:"projectIdSelector,omitempty"`
+
+	// New Deploy Key’s title
+	// required
+	Title string `json:"title,omitempty"`
```

Comment:
Required fields don't need `omitempty`
```suggestion
	Title string `json:"title"`
```